# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1272152916 14400
#      Sat Apr 24 19:48:36 2010 -0400
# Node ID 79410269cd21dfe6714dba5a6f5d25d43a7b09df
# Parent  5475da58d39fd1f6117366bcb49c348260b6ae1d
print.m: Set figure color property to 'none' when printing.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-24  Ben Abbott <bpabbott@mac.com>
+
+	* plot/print.m: Set figure color property to 'none' when printing.
+
 2010-04-24  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Correct fill of diamond markers
 
 2010-04-24  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__go_draw_axes__.m: Consistent placement of axes for ps vs
 	other gnuplot terminals.
diff --git a/scripts/plot/print.m b/scripts/plot/print.m
--- a/scripts/plot/print.m
+++ b/scripts/plot/print.m
@@ -581,16 +581,17 @@ function print (varargin)
 
     is_eps_file = strncmp (dev, "eps", 3);
     p.units = get (gcf, "units");
     p.paperunits = get (gcf, "paperunits");
     p.papersize = get (gcf, "papersize");
     p.paperposition = get (gcf, "paperposition");
     p.paperpositionmode = get (gcf, "paperpositionmode");
     p.paperorientation = get (gcf, "paperorientation");
+    p.color = get (gcf, "color");
     if (p.papersize(1) > p.papersize(2))
       paperorientation = "landscape";
     else
       paperorientation = "portrait";
     endif
     if (! strcmp (paperorientation, get (gcf, "paperorientation"))
         && warn_on_inconsistent_orientation)
        msg = {"print.m - inconsistent papersize and paperorientation properties.\n",
@@ -632,16 +633,17 @@ function print (varargin)
     if (! isfield (figure_properties, "__pixels_per_inch__"))
       addproperty ("__pixels_per_inch__", gcf, "double", resolution);
     endif
     set (gcf, "__pixels_per_inch__", resolution)
 
     unwind_protect
       set (gcf, "paperunits", "inches");
       set (gcf, "units", "pixels");
+      set (gcf, "color", "none");
       restore_properties = true;
       if ((! output_for_printer || is_eps_file) && ! doprint)
         ## If not PDF or PostScript, and the result is not being sent to a printer,
         ## render an image the size of the paperposition box.
         ## Trigger the listener to convert all paper props to inches.
         if (! isempty (canvas_size))
           size_in_pixels = sscanf (canvas_size ,"%d, %d");
           size_in_pixels = reshape (size_in_pixels, [1, numel(size_in_pixels)]);
